Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -201,11 +204,46 @@ namespace AZ::Render
         return AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
     }
 
-    void AtomActorDebugDraw::RenderAABB(EMotionFX::ActorInstance* instance, const AZ::Color& aabbColor)
+    void AtomActorDebugDraw::RenderAABB(EMotionFX::ActorInstance* instance,
+            bool enableNodeAabb,
+            const AZ::Color& nodeAabbColor,
+            bool enableMeshAabb,
+            const AZ::Color& meshAabbColor,
+            bool enableStaticAabb,
+            const AZ::Color& staticAabbColor)
     {
         RPI::AuxGeomDrawPtr auxGeom = m_auxGeomFeatureProcessor->GetDrawQueue();
-        const AZ::Aabb& aabb = instance->GetAabb();
-        auxGeom->DrawAabb(aabb, aabbColor, RPI::AuxGeomDraw::DrawStyle::Line);
+
+        if (enableNodeAabb)
+        {
+            AZ::Aabb aabb;
+            instance->CalcNodeBasedAabb(&aabb);
+            if (aabb.IsValid())
+            {
+                auxGeom->DrawAabb(aabb, nodeAabbColor, RPI::AuxGeomDraw::DrawStyle::Line);
+            }
+        }
+
+        if (enableMeshAabb)
+        {
+            AZ::Aabb aabb;
+            const size_t lodLevel = instance->GetLODLevel();
+            instance->CalcMeshBasedAabb(lodLevel, &aabb);
+            if (aabb.IsValid())
+            {
+                auxGeom->DrawAabb(aabb, meshAabbColor, RPI::AuxGeomDraw::DrawStyle::Line);
+            }
+        }
+
+        if (enableStaticAabb)
+        {
+            AZ::Aabb aabb;
+            instance->CalcStaticBasedAabb(&aabb);
```

Comment:
We don't need to recalculate the static AABB each frame. This is calculated after loading finished and does not change. Can you use `GetStaticAabb()` instead please?